### PR TITLE
Add test workflox and tox

### PR DESCRIPTION
**Pull request by CaselIT**

### CODE REVIEW

**Review by zzzeek — APPROVED** [review]
LGTM

File: .github/workflows/run-test.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+env:
+  # global env to all steps
+  TOX_WORKERS: -n2
+
+jobs:
+  run-test:
+    name: ${{ matrix.python-version }}-${{ matrix.sqlalchemy }}
+    runs-on: "ubuntu-latest"
+    strategy:
+      # run this job using this matrix, excluding some combinations below.
```
Comment (ashb) [inline .github/workflows/run-test.yaml:16]:
```suggestion
```

File: .github/workflows/run-test.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    # steps to run in each job. Some are github actions, others run shell commands
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Set up python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: ${{ matrix.architecture }}
```
Comment (ashb) [inline .github/workflows/run-test.yaml:35]: I don't see architecture in the matrix?

Comment (CaselIT) [reply .github/workflows/run-test.yaml:34]: indeed it's not, but it works in any case.

it's probably a leftover from a file I used as templace

**Comment by CaselIT** [comment]
gonna merge this, we can tune it later if needed

**Pull request merged**